numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1482,7 +1482,7 @@ arr_add_docstring(PyObject *NPY_UNUSED(dummy), PyObject *args)
         if (!(doc)) {                                                   \
```

Comment:
just a nit and unrelated to this change, but the do {...} while(0) can probably be removed. 
another nit and again unrelated to this change, but when i read this, i was searching for new in the code, but didnt realize that new was defined inside the macro, which made this a bit confusing.

Reply:
Yeah, lets move the new in front of the macro, also removes the need for the macro cast...

Reply:
I tried to clean it up, which deletes a bunch of other code since Python seems to define those types in the header now (if not, I assume CI will fail).